### PR TITLE
supervisor edit subject

### PR DESCRIPTION
**Pull request by nguyenngocha**

### CODE REVIEW

File: app/controllers/supervisor/subjects_controller.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ def index
     subject = Subject.new
     subject.tasks.build
     @subject_form = SubjectForm.new subject
+    @title = t("subject.new_subject")
```
Comment (kagoromo) [inline app/controllers/supervisor/subjects_controller.rb]: ()

File: app/controllers/supervisor/subjects_controller.rb [hunk 2]
```diff
@@ -24,6 +25,24 @@ def create
     end
   end
 
+  def edit
+    @subject_form = SubjectForm.new Subject.find params[:id]
+    @subjects = Subject.all.page(params[:page]).per Settings.subjects.per_page
+    @title = t("subject.edit_subject")
```
Comment (kagoromo) [inline app/controllers/supervisor/subjects_controller.rb]: ()

**Comment by kagoromo** [comment]
OK

File: app/controllers/supervisor/subjects_controller.rb [hunk 3]
```diff
@@ -24,6 +25,24 @@ def create
...
+    render :index
+  end
+
+  def update
+    @subject_form = SubjectForm.new Subject.find params[:id]
+    if @subject_form.validate params[:subject].permit!
+      @subject_form.save
+      flash[:success] = t "subject.edit_success"
+    else
+      flash.now[:danger] = t "subject.edit_fail"
```
Comment (kagoromo) [inline app/controllers/supervisor/subjects_controller.rb]: render

**Comment by kagoromo** [comment]
OK

**Comment by kagoromo** [comment]
OK

File: config/routes.rb [hunk 4]
```diff
@@ -15,7 +15,7 @@
     end
     resources :user_courses, only: :destroy
     resources :users
-    resources :subjects, only:[:index, :create]
+    resources :subjects, only:[:index, :create, :edit, :update]
```
Comment (hellovietnam93) [inline config/routes.rb]: sử dung except nhé,. chú ý convention

**Pull request merged**